src/test/java/org/folio/search/controller/SearchInstanceIT.java

```diff
@@ -2,119 +2,63 @@
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id", is(getSemanticWeb().getId())));
-  }
-
-  @Test
-  void canSearchByPublisher_byWord() throws Exception {
-    mockMvc.perform(get(searchInstancesByQuery("publisher all {value}"), "press")
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id", is(getSemanticWeb().getId())));
-  }
-
-  @Test
-  void canSearchByContributorsName() throws Exception {
-    mockMvc.perform(get(searchInstancesByQuery("contributors.name all {value}"), "frank")
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id",
-        is(getSemanticWeb().getId())))
-      .andExpect(jsonPath("instances[0].contributors[0].name",
-        is("Antoniou, Grigoris")))
-      .andExpect(jsonPath("instances[0].contributors[1].name",
-        is("Van Harmelen, Frank")));
-  }
-
-  @Test
-  void canSearchByContributors() throws Exception {
-    mockMvc.perform(get(searchInstancesByQuery("contributors all {value}"), "grigoris")
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id", is(getSemanticWeb().getId())));
+  private static Stream<Arguments> positiveSearchTestDataProvider() {
+    return Stream.of(
+      arguments("search by instance id for exactMatch", "id={value}", array(getSemanticWeb().getId()), null),
+      arguments("search by instance id using wildcard", "id={value}", array("5bf370e0*a0a39"), null),
+      arguments("search by instance title (title)", "title all {value}", array("semantic"), null),
+      arguments("search by instance title (series)", "title all {value}", array("cooperative"), null),
+      arguments("search by instance title (series, partial)", "title all {value}", array("cooperate"), null),
+      arguments("search by instance title (part of title)", "title all {value}", array("primers"), null),
+      arguments("search by instance title (alternative title)", "title all {value}", array("primers"), null),
+      arguments("search by publisher (abbreviate)", "publisher all {value}", array("MIT"), null),
+      arguments("search by publisher (abbreviate lowercase)", "publisher all {value}", array("mit"), null),
+      arguments("search by publisher (word)", "publisher all {value}", array("press"), null),
+      arguments("search by contributors name", "contributors.name all {value}", array("frank"),
+        (ThrowingConsumer<ResultActions>) searchResult -> searchResult
+          .andExpect(jsonPath("instances[0].contributors[0].name", is("Antoniou, Grigoris")))
+          .andExpect(jsonPath("instances[0].contributors[1].name", is("Van Harmelen, Frank")))),
```

Comment:
Feel free to remove it, It is covered by a separate test.

Reply:
I will leave it as example for future development, this check does not takes a lot of time, but can be handful in future cases with sorting and so on